lib/orocos/process.rb

```diff
@@ -1190,7 +1220,7 @@ def self.wait_running(process, timeout = nil, name_service = Orocos::CORBA.name_
                 start_time = Time.now
                 got_alive = process.alive?
                 loop do
-                    break if wait_running(process, 0, name_service, &block)
```

Comment:
What you could (should ?) do here is IO.select([ior_read], $ADD_TIMEOUT) to wait for new data on the pipe, instead of sleeping, now that we have an explicit channel for that.